.github/workflows/test-suite.yml

```diff
@@ -8,17 +8,17 @@ jobs:
     strategy:
       max-parallel: 5
       matrix:
-        python: [3.7, 3.8]
+        python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

Comment:
Why quoting?
As for the confusing error message, it is a feature of YAML parsing. When you specify input like
```yaml
version: 3.10
```
3.10 is parsed as a number and it is trimmed to 3.1 that looks like expected behavior for numbers.
You should specify input with quotes to treat it as string.
```yaml
version: '3.10'
```
https://github.com/actions/setup-python/issues/160#issuecomment-724485470
https://github.com/actions/runner/issues/1989

Reply:
I got bit by this many times, trying to install Python 3.1. It's very annoying.